plugins/janus_videoroom.c

```diff
@@ -2854,30 +2948,60 @@ static json_t *janus_videoroom_process_synchronous_request(janus_videoroom_sessi
 			if(!ok) {
 				JANUS_LOG(LOG_ERR, "Invalid element in the allowed array (not a string)\n");
 				error_code = JANUS_VIDEOROOM_ERROR_INVALID_ELEMENT;
-				g_snprintf(error_cause, 512, "Invalid element in the allowed array (not a string)");
+				g_snprintf(error_cause, JANUS_ERROR_CAUSE_STRING_SIZE, "Invalid element in the allowed array (not a string)");
 				goto prepare_response;
 			}
 		}
 		gboolean save = permanent ? json_is_true(permanent) : FALSE;
 		if(save && config == NULL) {
 			JANUS_LOG(LOG_ERR, "No configuration file, can't create permanent room\n");
 			error_code = JANUS_VIDEOROOM_ERROR_UNKNOWN_ERROR;
-			g_snprintf(error_cause, 512, "No configuration file, can't create permanent room");
+			g_snprintf(error_cause, JANUS_ERROR_CAUSE_STRING_SIZE, "No configuration file, can't create permanent room");
 			goto prepare_response;
 		}
 		guint64 room_id = 0;
 		char room_id_num[30], *room_id_str = NULL;
 		json_t *room = json_object_get(root, "room");
-		if(!string_ids) {
+                if(room) { /*CARBYNE-LOGIC*/
+		   if(!string_ids) {
 			room_id = json_integer_value(room);
 			g_snprintf(room_id_num, sizeof(room_id_num), "%"SCNu64, room_id);
 			room_id_str = room_id_num;
-		} else {
+		    } else {
 			room_id_str = (char *)json_string_value(room);
-		}
-		if(room_id == 0 && room_id_str == NULL) {
+		      }
+		    if(room_id == 0 && room_id_str == NULL) {
 			JANUS_LOG(LOG_WARN, "Desired room ID is empty, which is not allowed... picking random ID instead\n");
-		}
+                        error_code = JANUS_VIDEOROOM_ERROR_UNKNOWN_ERROR; /*CARBYNE-AUT*/
+                        g_snprintf(error_cause, JANUS_ERROR_CAUSE_STRING_SIZE, "Desired room ID is 0, which is not allowed"); /*CARBYNE-AUT*/
+                        goto prepare_response;/*CARBYNE-AUT*/
+		   }
+                } else { /*CARBYNE-LOGIC*/
+                     error_code = JANUS_VIDEOROOM_ERROR_UNKNOWN_ERROR;
+                     g_snprintf(error_cause, JANUS_ERROR_CAUSE_STRING_SIZE, "room is mandatory for this version");
+                     goto prepare_response;
+                }  /*CARBYNE-LOGIC*/
+
+                /*CARBYNE-AUT */
+                if(auth_secret != NULL) {
+                    JANUS_VALIDATE_JSON_OBJECT(root, wgw_token_auth_parameters,
+                                               error_code, error_cause, TRUE,
+                                JANUS_VIDEOROOM_ERROR_MISSING_ELEMENT, JANUS_VIDEOROOM_ERROR_INVALID_ELEMENT);
+                    if(error_code != 0)
+                         goto prepare_response;
+
+                    json_t *vr_token = json_object_get(root, "roomtoken");
+                    const char *vr_token_text = json_string_value(vr_token);
+
+                    JANUS_LOG(LOG_INFO, " create:plugin token:  %s\n", vr_token_text);
+                    if(FALSE ==  janus_auth_check_signature(vr_token_text,(const char *)room_id_str)) {
+                       error_code = JANUS_VIDEOROOM_ERROR_UNKNOWN_ERROR;
```

Comment:
Replace JANUS_VIDEOROOM_ERROR_UNKNOWN_ERROR with the actual error - invalid token